primefaces/src/main/java/org/primefaces/csp/CspPhaseListener.java

```diff
@@ -67,7 +70,16 @@ public void beforePhase(PhaseEvent event) {
         CspState state = PrimeFacesContext.getCspState(context);
 
         String policy = LangUtils.isBlank(customPolicy.get()) ? "script-src 'self'" : customPolicy.get();
-        externalContext.addResponseHeader("Content-Security-Policy", policy + " 'nonce-" + state.getNonce() + "'");
+        policy += " 'nonce-" + state.getNonce() + "';";
```

Comment:
i think this var should be moved in the `else` statement?

Reply:
Agreed.  Moved

`